network/peer/peer.go

```diff
@@ -702,7 +706,114 @@ func (p *peer) handle(msg message.InboundMessage) {
 	p.Router.HandleInbound(context.Background(), msg)
 }
 
-func (p *peer) handlePing(*p2p.Ping) {
+func (p *peer) handlePing(msg *p2p.Ping) {
+	// if GetUptime is not 0 then this is a guaranteed new ping message
+	// It can be 0 in both cases: 1) old ping message
+	// 2) new ping message and we're not a validator (or an actual 0 uptime)
+	// In both cases we should handle it as an old ping message since
+	// it is backwards compatible
+	// TODO: remove this in the future
+	if msg.GetUptime() != 0 {
+		p.handleNewPing(msg)
+	} else {
+		p.handleOldPing(msg)
+	}
+}
+
+func (p *peer) createPingMessage() (message.OutboundMessage, error) {
+	idList := p.trackedSubnets.List()
+	subnetUptimes := make([]*p2p.SubnetUptime, 0, len(idList))
+
+	for _, subnetID := range idList {
+		subnetUptime, err := p.UptimeCalculator.CalculateUptimePercent(p.id, subnetID)
+		if err != nil {
+			p.Log.Debug("failed to get peer uptime percentage",
+				zap.Stringer("nodeID", p.id),
+				zap.Stringer("subnetID", subnetID),
+				zap.Error(err),
+			)
+			continue
+		}
+
+		copyID := subnetID
```

Comment:
We typically write this as:
```suggestion
		subnetID := subnetID
```
in other places

Reply:
done